src/Identity/Extensions.Core/src/UserManager.cs

```diff
@@ -2276,7 +2277,60 @@ public virtual string GenerateNewAuthenticatorKey()
     /// </summary>
     /// <returns></returns>
     protected virtual string CreateTwoFactorRecoveryCode()
-        => Guid.NewGuid().ToString().Substring(0, 8);
+    {
+        var recoveryCode = new StringBuilder(11);
```

Comment:
Can we param the size and do two loops that `recoveryCode.Append(GetRandomRecoveryCodeChar())` n/2 times?

Reply:
You mean like call it with two n=5 for loops with a '-' in between?

Reply:
Yeah, unless you think that's uglier

Reply:
I would keep it like this